Wikipedia/Code/EmptyRefreshingViewController.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+protocol EmptyRefreshingViewControllerDelegate: class {
+    func triggeredRefresh(refreshCompletion: @escaping () -> Void)
+}
+
+class EmptyRefreshingViewController: ViewController {
+    private let refreshControl = UIRefreshControl()
+    private var emptyView: WMFEmptyView? = nil
+    weak var delegate: EmptyRefreshingViewControllerDelegate?
+    
+    var type: WMFEmptyViewType? {
+        didSet {
+            if oldValue != type {
+                emptyView?.removeFromSuperview()
+                emptyView = nil
+                
+                if let newType = type,
+                    let emptyView = EmptyRefreshingViewController.emptyView(of: newType, theme: self.theme, frame: view.bounds) {
+                    scrollView?.addSubview(emptyView)
+                    self.emptyView = emptyView
+                }
+            }
+        }
+    }
+    
+    override init() {
+
+        super.init()
+        
+        let scrollView = UIScrollView()
+        scrollView.alwaysBounceVertical = true
+        view.wmf_addSubviewWithConstraintsToEdges(scrollView)
+        self.scrollView = scrollView
+        
+        refreshControl.layer.zPosition = -100
+        refreshControl.addTarget(self, action: #selector(refreshControlActivated), for: .valueChanged)
+        
+        scrollView.addSubview(refreshControl)
+        scrollView.refreshControl = refreshControl
+        
+    }
+    
+    @objc private func refreshControlActivated() {
+        
+        delegate?.triggeredRefresh(refreshCompletion: { [weak self] in
+            self?.refreshControl.endRefreshing()
+        })
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func apply(theme: Theme) {
+        super.apply(theme: theme)
```

Comment:
should this guard against view not being loaded like the parent does? 
```
        self.theme = theme
        guard viewIfLoaded != nil else {
            return
        }
```

Reply:
@nambatee done, added this on all the talk page related classes I missed this on 😬. It caused a couple of bugs that I fixed as a part of the PR feedback commit too.